corehq/form_processor/document_stores.py

```diff
@@ -65,11 +65,14 @@ def get_document(self, doc_id, *, external_id=None):
             if external_id is None:
                 return CommCareCase.objects.get_case(doc_id, self.domain).to_json()
             else:
-                return CommCareCase.objects.get_case_by_external_id(
+                case = CommCareCase.objects.get_case_by_external_id(
                     self.domain,
                     external_id,
                     raise_multiple=True
-                ).to_json()
+                )
+                if case:
+                    return case.to_json()
+                raise CaseNotFound(f'external_id: {external_id!r}')
```

Comment:
It calls `repr()` on a variable instead of `str()`. So for strings (which is what external_id should be because it's stored in a CharField) it's the same as `f"external_id: '{external_id}'"`, but in general it gives the user more information about a variable than just `str()`, so that's why I use it in error messages.

Reply:
okay. 
nit: I would think of `repr` a internal use like in log/debugging? and not user facing though

Reply:
I agree.

Exception messages should be internal -- this string is not translated, and if it does reach the browser, it'll just be a Sad Danny, won't it?

Reply:
Sorry for the pushback. I'm sure the project wants to get this merged. https://github.com/dimagi/commcare-hq/pull/35007/commits/087cd322b8a36077e8717902df18a31a3c2064ac